src/stores/spaces/SpaceStore.ts

```diff
@@ -1071,6 +1079,9 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
         }
     }
 
+    // TODO: room type, Once it is possible to update/change the room type of a room we need to add a `onRoomTypeChange` callback here
+    // to update the `SpaceStore` accordingly.
```

Comment:
You need an onRoomTypeChange anyway, as the room type may arrive down `/sync` late, there's no guaranteed atomicity.

Reply:
oh interesting. Even `m.room.create` state event gets synced lazily?

Reply:
Yes, nothing guarantees the `m.room.create` event will be sent in the first batch

Reply:
I added a room update on create event.